### PR TITLE
Record sorting

### PR DESCRIPTION
**Pull request by EmanH**
- Changed sort column operation to `RequestColumnSort`. Sorting should be done outside of the grid as worker threads are not supported in an Angular library due [to limitations](https://github.com/angular/angular-cli/issues/15059)
- Support multi-column sort indicators in column headers

### CODE REVIEW

**Comment by cli2022** [comment]
@EmanH Probably the version of `ngx-grid-core` should be incremented?

**Comment by EmanH** [comment]
> @EmanH Probably the version of `ngx-grid-core` should be incremented?

@cli2022 Yes, I forgot to push the commit earlier :)

File: projects/ngx-grid-core/src/lib/grid-data-source.ts [hunk 1]
```diff
@@ -115,10 +115,12 @@ export class GridDataSource implements IGridDataSource {
     ))
   }
 
+  upsertRows(rows: IGridRow[]): IGridRow[]
   upsertRows(...rows: IGridRow[]): IGridRow[]
   upsertRows(index: number, ...rows: IGridRow[]): IGridRow[]
-  upsertRows(param1: number | IGridRow, ...rows: IGridRow[]): IGridRow[] {
-    if (typeof param1 === 'object') rows.unshift(param1)
+  upsertRows(param1: number | IGridRow | IGridRow[], ...rows: IGridRow[]): IGridRow[] {
```
Comment (cli2022) [inline projects/ngx-grid-core/src/lib/grid-data-source.ts:121]: Maybe should consider to rename `param1` to use better naming.

File: projects/ngx-grid-core/src/lib/ui/header/header.component.ts [hunk 2]
```diff
@@ -83,7 +78,15 @@ export class HeaderComponent extends AutoUnsubscribe implements OnInit {
...
   public endResize   = () => setTimeout(() => this._isResizing = false)
-  public sortColumn  = (i: number) => !this._isResizing && this.gridController.column.SortColumn.run(this.columns.value[i])
+  public sortColumn  = (i: number) => {
+    if (this._isResizing) return
+    const columnKey = this.columns.value[i]
+    const sortOrder = this.currentSortOrder(columnKey)
+    // Run the column operation RequestColumnSort 
+    // toggle sort direction
+    // asc -> desc -> natural
+    this.gridController.column.RequestColumnSort.run(columnKey, sortOrder === ESortDirection.Asc ? ESortDirection.Desc : sortOrder === ESortDirection.Desc ? ESortDirection.Natural : ESortDirection.Asc)
```
Comment (cli2022) [inline projects/ngx-grid-core/src/lib/ui/header/header.component.ts:88]: The design is currently to toggle between Asc and Desc sorting order, wonder what is the purpose of toggling from Desc to Natural?

Comment (cli2022) [reply projects/ngx-grid-core/src/lib/ui/header/header.component.ts:88]: Alright, after discussion, I think we are happy to go with toggling between three states.

**Review by cli2022 — APPROVED** [review]
@EmanH The changes look good.

**Pull request merged**